spec/controllers/api/v1/registrations_controller_spec.rb

```diff
@@ -0,0 +1,90 @@
+# frozen_string_literal: true
+
+require 'rails_helper'
+
+RSpec.describe API::V1::RegistrationsController, type: :controller do
+  describe 'POST#create', devise_mapping: true do
+    context 'given an invalid oauth application' do
+      it 'returns forbidden status' do
+        params = { user: { email: 'test@test.test', password: 'test@test.test', password_confirmation: 'test@test.test' } }
+        post :create, params: params
+
+        expect(response).to have_http_status(:forbidden)
+      end
+    end
+
+    context 'given a valid oauth application' do
```

Comment:
This case should come first instead of `given an invalid oauth application` case as we write happy case first

Reply:
Good point! Will update this